o4/o4.py

```diff
@@ -955,21 +986,49 @@ def move_except(from_dir, to_dir, but_not):
 
 def o4_fail():
     files = []
+    passthroughs = []
     n = 0
-    for f in map(fstat_split, sys.stdin):
+    for line in sys.stdin:
+        if line.startswith('#o4pass-'):
+            msgtype, _, msg = line.replace('#o4pass-', '').partition('#')
+            if msgtype == 'warn':
+                passthroughs.append(msg)
+            continue
+        f = fstat_split(line)
         if not f:
             continue
         n += 1
         if n < 100:
             files.append(f"  {f[F_PATH]}#{f[F_REVISION]}")
+
+    if not files and not passthroughs:
+        sys.exit(0)
+
+    err_print()
+    hdr = ' o4 ERROR ' if files else ' o4 WARNING '
+    l = (78 - len(hdr)) // 2
+    hdr = '*' * l + hdr + '*' * l
+    ftr = '*' * len(hdr)
+    err_print(f'{CLR}{hdr}')
+
     if files:
-        dash = '-' * 52
-        err_print(f"\n{CLR}*** ERROR: These files did not get synced correctly:\n{dash}")
+        err_print('These files did not get processed by the o4 pipeline:')
         err_print('\n'.join(sorted(files)))
         if len(files) != n:
-            err_print(f"  ...and {n-len(files)} others!")
-        err_print(dash)
-        sys.exit(f"{CLR}*** ERROR: Pipeline ended with {n} fstat lines.")
+            err_print(f'  ...and {n-len(files)} others!')
+        err_print(f'\n{ftr}')
+
+    if passthroughs:
+        err_print('\n'.join(sorted(passthroughs)))
+        err_print(ftr)
+        if not files:
+            with open(INCOMPLETE_INDICATOR, 'w') as f:
+                pass
+            err_print(f'{CLR} o4 IS SUCCEEDING EVEN THOUGH SOME FILES ARE NOT UP-TO-DATE')
+            err_print(ftr)
+            sys.exit(0)
+
+    sys.exit(f"{CLR}*** ERROR: Pipeline ended with {n} fstat lines.")
```

Comment:
This is unrelated to the code change something for a TODO list. Is there a better error message than this technical description?